Функциональность «Общие фильмы»

AlksSrg opened the pull request:
Функциональность «Общие фильмы»
1)Правки по комментариям в методе getRecommendations класса UserDbService;
2)Доработка кода в классе LikeDao и LikeDaoImpl;
3)Реализация end-point @GetMapping("/common") класса FilmController который возвращает список общих фильмов для 2-ух пользователей;
4)Доработка класса FilmDbService - реализация метода который возвращает список общих фильмов для 2-ух;
5)Переписан код по удалению лайков;

AlksSrg committed:
Функциональность «Общие фильмы»
1)Правки по комментариям в методе getRecommendations класса UserDbService;
2)Доработка кода в классе LikeDao и LikeDaoImpl;
3)Реализация end-point @GetMapping("/common") класса FilmController который возвращает список общих фильмов для 2-ух пользователей;
4)Доработка класса FilmDbService - реализация метода который возвращает список общих фильмов для 2-ух;
5)Переписан код по удалению лайков;

Irina-Belikova commented on src/main/java/ru/yandex/practicum/filmorate/service/FilmDbService.java:
```diff
@@ -216,4 +224,39 @@ public Collection<Film> getFilmsByUser(Long id) {
+
+        // Получаем фильмы, обогащаем и сортируем
+        return filmStorage.getFilmsByIds(commonFilmIds).stream()
+                .peek(this::enrichFilmWithDetails)
```
Ну мне все так же не нравятся запросы n+1 (строка 256). Но не настаиваю. В своих режиссерах сделаю наполнение фильмов за 1 запрос. В принципе, потом уже можно будет перенести это и сюда при желании. В остальном все ОК.

AlksSrg committed:
Функциональность «Общие фильмы»
1)Правки по методу getCommonFilms класса FilmDbService для увеличения скорости работы приложения;
